### ISSUE TITLE
Poor error span for unclosed JSX tags in the presence of whitespace/comments

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#37306]
```tsx
let Foo = {
  Bar() {
    
  }
}

let x = <    Foo.Bar >Hello

```

**Expected**: An error span like

```tsx
let x = <    Foo.Bar >Hello
             ~~~~~~~
```

**Actual**: An error span like
```tsx

let x = <    Foo.Bar >Hello
         ~~~~~~~~~~~
```

**Comment by AGMETEOR** [comment]
Would like to try this one out...

**Comment by DanielRosenwasser** [comment]
Go for it! I'd try running the change to see if any tests are affected first, and if not, add the one in the example in a a file named `tests/cases/compiler/errorSpanForUnclosedJsxTag.tsx`

**Comment by AGMETEOR** [comment]
@DanielRosenwasser implemented a solution for this. Will be raising a PR for review shortly :)

**Comment by AGMETEOR** [comment]
Hi @DanielRosenwasser , I opened a PR for this issue :)

**Comment by DanielRosenwasser** [comment]
Thanks @AGMETEOR!
